5-add-owasp-dependency-check

ViktorKapustianyk opened the pull request:
build(gradle): integrate OWASP Dependency Check plugin

Added OWASP Dependency Check plugin version 11.1.0 to identify and manage known vulnerabilities in dependencies.

BREAKING CHANGE: This commit introduces a new plugin which may affect build configurations.

khshanovskyi reviewed (commented):
add such file to root
`dependency-check-suppression-file.xml`
with such content
`
<?xml version="1.0" encoding="UTF-8"?>

<!-- For examples on how to suppress false positives see documentation here:
     https://jeremylong.github.io/DependencyCheck/general/suppression.html -->

<suppressions xmlns="https://jeremylong.github.io/DependencyCheck/dependency-suppression.1.3.xsd">
    <!--   here will be suppressions in case if needed -->
</suppressions>
`

DimaGalchenko commented on build.gradle:
```diff
@@ -1,5 +1,6 @@
 plugins {
     id 'java'
+    id "org.owasp.dependencycheck" version "11.1.0"
```
Please, move plugin version to (https://github.com/DimaGalchenko/winter-framework/blob/develop/version.gradle) file

martaisty replied:
I am not sure that will work. `applyFrom: version.gradle` goes after plugins block

DimaGalchenko replied:
According to PR(adding Jacoco) it should works:
<img width="256" alt="Screenshot 2024-11-04 at 09 50 36" src="https://github.com/user-attachments/assets/ff238ba2-24c2-43be-9e91-64ceb3b27651">
But I am not sure

ViktorKapustianyk replied:
Added OWASP Dependency Check plugin to `build.gradle` using an externalized version reference in `version.gradle`. Updated the `plugins` block to dynamically load the version from the `versions` map in `version.gradle`.

ViktorKapustianyk commented:
build(dependency-check): integrate OWASP Dependency Check with suppression file

Added OWASP Dependency Check plugin to build.gradle for identifying known vulnerabilities in dependencies.
Configured suppression file (dependency-check-suppressions.xml) to handle any false positives.

Added the following changes:
- Included 'org.owasp.dependencycheck' plugin in build.gradle.
- Configured suppression file for dependency check in the project root.
- Updated documentation link for false positive suppression.

Additional updates:
- Addressed URI registration warning in IntelliJ for XML schema validation.

DimaGalchenko commented:
Please, resolve all conflicts
